metric/distribution/regular/regular_distribution.go

```diff
@@ -72,7 +72,7 @@ func (regularDist *RegularDistribution) AddEntryWithUnit(value float64, weight f
 	if weight <= 0 {
 		return fmt.Errorf("unsupported weight %v: %w", weight, distribution.ErrUnsupportedWeight)
 	}
-	if value < 0 || math.IsNaN(value) || math.IsInf(value, 0) {
+	if value < 0 || math.IsNaN(value) || math.IsInf(value, 0) || !distribution.IsValueInRange(value) {
```

Comment:
in range check is half redundant with the value < 0 check

Reply:
IsValueInRange could just take in the upper and lower bound, 0, x).
Or it could hard code the boundaries if we don't expect them to change.

Another option is to move all the conditions into a single function that checks value.

Reply:
That's fair. I wanted to keep it consistent, but I can change it.

Reply:
Moved it into a single `IsAcceptedValue` function.